### PR TITLE
fix: ValidatedTextInput  attribute

### LINKED ISSUE #51792: Accessibility: Error is not programmatically associated
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

An error is described with text, but it is not programmatically associated with its related field.
T001_SR_ErrorNotAssociated

### Component or Page: 

Page: Checkout Process (Block-Based)

### Name affected element:

1. First Name (Please enter a valid first name)
2. Last Name (Please enter a valid last name)
3. Address (Please enter a valid address)
4. City (Please enter a valid city)
5. Zip (Please enter a valid zip code)

### Relevant code snippets:

`<div class="wc-block-components-text-input wc-block-components-address-form__first_name has-error"><input type="text" id="shipping-first_name" autocapitalize="sentences" autocomplete="given-name" aria-label="First name" required="" aria-invalid="true" title="" value="" aria-describedby="validate-error-shipping_first_name"><label for="shipping-first_name">First name</label><div class="wc-block-components-validation-error" role="alert"><p>Please enter a valid first name</p></div></div>`

### Recommended fix:

Programmatically associate form fields with their respective errors. Use aria-errormessage to programmatically bind the form control with the related error. 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage

### PR DESCRIPTION
**Pull request by frosso**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/51792

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Build the project (`pnpm run build`)
- [Ensure your checkout page uses the checkout block
](https://woocommerce.com/document/woocommerce-store-editing/customizing-cart-and-checkout/#using-the-cart-and-checkout-blocks)
- Add a product to the cart (e.g.: the "Album" product from the product templates
- Go to the checkout page
- Don't fill up all the fields - leave last name empty, for example
- Try to submit the checkout form by clicking "Place order"
- An error should appear below the empty fields
- The error message should have an `id` attribute, whose value is referenced on the respective input, which also has the `aria-errormessage` attribute with the same value as the `id` attribute on the error message
Screenshot 2024-10-09 at 10 05 03 AM

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message fix: ValidatedTextInput aria-errormessage matching

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx [hunk 1]
```diff
@@ -232,10 +232,6 @@ const ValidatedTextInput = forwardRef<
 		}
 
 		const hasError = validationError?.message && ! validationError?.hidden;
-		const describedBy =
```
Comment (frosso) [inline plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx:235]: We don't want to overwrite the `aria-describedby` attribute when there is an error - the `aria-describedby` should not change when an error is present.

File: plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx [hunk 2]
```diff
@@ -271,7 +275,7 @@ const ValidatedTextInput = forwardRef<
 					}
 				} }
 				onBlur={ () => validateInput( false ) }
-				ariaDescribedBy={ describedBy }
+				ariaDescribedBy={ ariaDescribedBy }
```
Comment (frosso) [inline plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx:278]: See message above

**Comment by frosso** [comment]
I re-assigned, since Paulo is AFK :)

**Review by mikejolley — APPROVED** [review]
Thanks, this looks ready to go.

**Pull request merged**